### ISSUE TITLE
Scroll to first invalid element

### ISSUE THREAD

**Issue by berteruan** [Rocketseat/unform#104]
**What would you like to be added**:
 Scroll to first invalid form element when form is submitted

**Why is this needed**:
In large forms it is common the first fields are hidden when we submit the form, so that the user does not see the error.

**Is your enhancement request related to a problem? Please describe.**
No.

**Additional context**

**Comment by zaguiini** [comment]
I don't think that adding this would be a good idea because it's very specific.
A good idea would be "add hooks" to validation. For example:

```js
<Form
  onValidation={({ errors }) => {
    // and here we might have the fields with errors, so you can imperatively scroll to that field, for example
  }}
/>```

With this pattern, it would be way more generic and cleaner. Not sure about the method signature, though. @diego3g thoughts?

**Comment by xbral** [comment]
@berteruan Você conseguiu implementar algo para esta funcionalidade?

**Comment by edumoreira1506** [comment]
Hello @xbral I opened one pull request with this functionality, you can see [here](https://github.com/Rocketseat/unform/pull/171)

**Comment by diego3g** [comment]
Unform released version 2.0 🎉

And now, validation is out of the box, so you can validate all data using your favorite validation library and then, as you have access to errors, you can grab the input name and scroll to it using JavaScript `element.offset().top`.

Check the [validation guide](https://unform.dev/guides/validation) for more info.
